Wikipedia/View Controllers/Nearby/NearbyViewController.m

```diff
@@ -122,12 +113,50 @@ - (BOOL)prefersStatusBarHidden {
     return NO;
 }
 
-- (void)viewWillDisappear:(BOOL)animated {
-    [self.locationManager stopUpdatingLocation];
+- (void)viewWillAppear:(BOOL)animated {
+    [super viewWillAppear:animated];
+
+    self.skipStartUpdates = NO;
+    // Only begin updates if by the time the startUpdates selector below is performed
+    // self.skipStartUpdates is still NO - lets us add the nearby view controller
+    // somewhere and immediately cover it up without it starting to update locations
+    // and headings.
+    [self performSelector:@selector(startUpdates) withObject:nil afterDelay:0.5];
```

Comment:
I dislike this approach because it goes against two best practices:

- Keeping state to a minimum
- Avoiding delays if possible (i.e. race condition vulnerabilities)

This one of the things I was hoping to prevent by having data migration (and onboarding) short-circuit the application by preventing the main view controller from even being instantiated.  If we did that, we wouldn't need any of this logic, and everything can behave normally w/o worrying about some external state transition.  We might be able to work around it here, but there will be (if there aren't already) other things going on "behind the scenes" while we show on-boarding (or worse, data migration) that we'd be better off without.

Since this was an issue before this ticket, and this ticket was only about moving on-boarding out of webVC, we have the option to implement what I described above in a separate ticket.  Let me know if you'd prefer a separate ticket, and I can create one w/ acceptance criteria along these lines.